packages/dht/src/dht/store/StoreManager.ts

```diff
@@ -80,11 +79,11 @@ export class StoreManager {
         if (selfIsPrimaryStorer) {
             sortedList.addContact(new Contact(newNode))
             const sorted = sortedList.getContactIds()
-            // findIndex should never return -1 here because we just added the new node to the list
+            // findIndex returns -1 if the node did not fit the redundancyFactor closest nodes
             const index = findIndex(sorted, (nodeId) => (nodeId === newNodeId))
             // if new node is within the storageRedundancyFactor closest nodes to the data
             // do replicate data to it
-            if (index < this.config.redundancyFactor) {
+            if (index !== -1) {
```

Comment:
This could be just `sortedList.getContactIds().includes(newNodeId)`?  (and maybe rename `sortedList` to e.g. `reduncancyFactorNodes`)